security/s2a/internal/crypter/chachapoly.go

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package crypter
+
+import (
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"crypto/cipher"
+	"fmt"
+)
+
+// Supported key sizes in bytes.
+const (
+	chachaKeySize256 = 32
+)
+
+// chachapoly is the struct that holds an CHACHA-POLY cipher for the S2A AEAD crypter.
+type chachapoly struct {
+	aead cipher.AEAD
+	// keySize stores the size of the key which was initially used. This
+	// is necessary to restrict key updates to the same key length as the
+	// initial key.
+	keySize int
+}
+
+// newCHACHAPOLY creates an CHACHA-POLY crypter instance. Note that the key must be
+// 256 bits.
```

Comment:
s/256 bits/32 bytes

Reply:
Since all of our data is in bytes (and not bits), we should make all references in bytes.